interface/modules/custom_modules/oe-module-comlink-telehealth/src/Controller/TeleconferenceRoomController.php

```diff
@@ -0,0 +1,786 @@
+        } else if ($action == 'check_registration') {
+            return $this->checkRegistrationAction($queryVars);
+        } else if ($action == 'get_telehealth_settings') {
+            return $this->getTeleHealthFrontendSettingsAction($queryVars);
+        } else {
+            $this->logger->error(self::class . '->dispatch() invalid action found', ['action' => $action]);
+            echo "action not supported";
+            return;
+        }
+    }
+
+    public function setAppointmentStatusAction($queryVars)
+    {
+        $pc_eid = $queryVars['pc_eid'];
+        $status = $queryVars['status'];
+        $authUser = $queryVars['authUser'];
+
+        try {
+            $userRepo = new UserService();
+            $user = $userRepo->getUserByUsername($authUser);
+
+            $appt = $this->appointmentService->getAppointment($pc_eid);
+
+            if (empty($appt)) {
+                throw new InvalidArgumentException("Invalid appointment pc_eid.  Appointment does not exist");
+            }
+
+            if ($this->isPatient) {
+                // throw error
+                throw new InvalidArgumentException("Cannot update appointment status in a patient context");
+            }
+            if (!AclMain::aclCheckCore('patients', 'appt', '', array('write','wsome'))) {
+                throw new AccessDeniedException("patients", "appt", "Does not have ACL permission to update appointment status");
+            }
+
+            $session = $this->sessionRepository->getSessionByAppointmentId($pc_eid);
+            if (empty($session)) {
+                // throw an error
+                throw new InvalidArgumentException("Invalid appointment eid");
+            }
+            if ($session['user_id'] != $user['id']) {
+                // throw an error
+                throw new AccessDeniedException("patients", "appt", "Cannot update appointment that belongs to different user");
+            }
+
+//            if (!$this->appointmentService->isValidAppointmentStatus($status)) {
+            if (!$this->isValidAppointmentStatus($status)) {
+                throw new InvalidArgumentException("Invalid appointment status received");
+            }
+            $this->appointmentService->updateAppointmentStatus($pc_eid, $status, $session['user_id'], $session['encounter']);
+            $this->logger->debug(
+                "TeleconferenceRoomController->setAppointmentStatusAction() Updated appointment status",
+                ['pc_eid' => $pc_eid, 'status' => $status, 'authUser' => $authUser]
+            );
+            echo text(json_encode(['status' => 'success']));
+        } catch (InvalidArgumentException $exception) {
+            (new SystemLogger())->errorLogCaller($exception->getMessage(), ['trace' => $exception->getTraceAsString(),
+                'queryVars' => $queryVars]);
+            http_response_code(400);
+            echo text(json_encode(['error' => 'invalid argument sent.  Check server logs for details']));
```

Comment:
no need to escape hard-coded strings (ie. without variables)
just seems like we are daring php to mess with us :)
lines 181, 186, 191, 196, 224, 252, 257, 304, 335, 473, 478,

Reply:
K fixed.